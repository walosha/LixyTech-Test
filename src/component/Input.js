import React from "react";

export default function Input({ onchange, error }) {
  return (
    <div
      style={{
        marginBottom: 20,
        display: "flex",
        flexDirection: "column",
        alignSelf: "center",
      }}
    >
      <input type="text" onChange={onchange} />
      {error && (
        <p style={{ color: "red" }}>
          This is not a valid Number, please enter a valid number!
        </p>
      )}
    </div>
  );
}
