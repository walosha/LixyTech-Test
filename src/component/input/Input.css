.inputContainer {
  margin-bottom: 20px;
  display: flex;
  flex-direction: column;
  align-self: center;
}

.error {
  color: red;
}
