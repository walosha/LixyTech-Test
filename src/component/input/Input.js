import React from "react";
import "./Input.css";

export default function Input({ onchange, error }) {
  return (
    <div className="inputContainer">
      <input type="text" onChange={onchange} />
      {error ? (
        <p className="error">
          This is not a valid Number, please enter a valid number!
        </p>
      ) : null}
    </div>
  );
}
