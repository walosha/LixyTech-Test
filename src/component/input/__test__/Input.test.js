import React from "react";
import ReactDom from "react-dom";
import { render } from "@testing-library/react";
import TestRenderer from "react-test-renderer";

import Input from "../Input";

it("Render Button Successfully ", () => {
  const div = document.createElement("div");
  ReactDom.render(<Input />, div);
});

it("Render Button Successfully ", () => {
  const tree = TestRenderer.create(<Input />).toJSON();
  expect(tree).toMatchSnapshot();
});
