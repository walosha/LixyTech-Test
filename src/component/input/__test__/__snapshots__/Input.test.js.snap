// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button Successfully  1`] = `
<div
  className="inputContainer"
>
  <input
    placeholder="Enter a number"
    type="text"
  />
</div>
`;
